src/storage/src/hummock/compactor/compactor_runner.rs

```diff
@@ -0,0 +1,117 @@
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use bytes::Bytes;
+use itertools::Itertools;
+use risingwave_hummock_sdk::key_range::KeyRange;
+use risingwave_pb::hummock::{CompactTask, LevelType};
+
+use crate::hummock::compactor::{CompactOutput, CompactionFilter, Compactor, CompactorContext};
+use crate::hummock::iterator::{
+    ConcatSstableIterator, Forward, HummockIterator, UnorderedMergeIteratorInner,
+};
+use crate::hummock::sstable::SstableIteratorReadOptions;
+use crate::hummock::utils::can_concat;
+use crate::hummock::{CachePolicy, CompressionAlgorithm, HummockResult, SstableBuilderOptions};
+
+#[derive(Clone)]
+pub struct CompactorRunner {
+    compact_task: CompactTask,
+    compactor: Compactor,
+}
+
+impl CompactorRunner {
+    pub fn new(context: Arc<CompactorContext>, task: CompactTask) -> Self {
+        let max_target_file_size = context.options.sstable_size_mb as usize * (1 << 20);
+        let cache_policy = if task.target_level == 0 {
+            CachePolicy::Fill
+        } else {
+            CachePolicy::NotFill
+        };
+        let mut options: SstableBuilderOptions = context.options.as_ref().into();
+        options.capacity = std::cmp::min(task.target_file_size as usize, max_target_file_size);
+        options.compression_algorithm = match task.compression_algorithm {
+            0 => CompressionAlgorithm::None,
+            1 => CompressionAlgorithm::Lz4,
+            _ => CompressionAlgorithm::Zstd,
+        };
+        let compactor = Compactor::new(
+            context.clone(),
+            options,
+            context.memory_limiter.clone(),
+            task.splits
+                .iter()
+                .map(|split| KeyRange {
+                    left: Bytes::copy_from_slice(split.get_left()),
+                    right: Bytes::copy_from_slice(split.get_right()),
+                    inf: split.get_inf(),
+                })
+                .collect_vec(),
+            cache_policy,
+            task.gc_delete_keys,
+            task.watermark,
+        );
+        Self {
+            compactor,
+            compact_task: task,
+        }
+    }
+
+    pub async fn run(
+        &self,
+        split_index: usize,
+        compaction_filter: impl CompactionFilter,
+    ) -> HummockResult<CompactOutput> {
+        let iter = self.build_sst_iter()?;
+        self.compactor
+            .compact_key_range_impl(split_index, iter, compaction_filter)
+            .await
+    }
+
+    /// Build the merge iterator based on the given input ssts.
+    fn build_sst_iter(&self) -> HummockResult<impl HummockIterator<Direction = Forward>> {
```

Comment:
For me this function is only strongly correlated with table_iter.

IMHO, the commets describe that `merge iterator based on the given input ssts`. hence, i am not sure that its a good way to transfer `table_infos` directly (may be it can more eary to test)

Reply:
I just use the existed merge iterator to avoid duplicate code...